src/cache/impl/LevelDbCache.ts

```diff
@@ -136,7 +136,37 @@ export class LevelDbCache<V = any> implements SortKeyCache<V> {
     return this.db as S;
   }
 
+  async getNumEntries(): Promise<Number> {
+    const keys = await this.db.keys().all();
+    return keys.length
+  }
+
   async prune(entriesStored = 5): Promise<PruneStats> {
-    throw new Error('Not implemented yet');
+    if (!entriesStored || entriesStored <= 0) {
+      entriesStored = 1
+    }
+
+    const keys = await this.db.keys().all();
+    for (const key of keys) {
+      const contractTxId = key.substring(1, 44)
+      const contractCache = this.db.sublevel<string, any>(contractTxId, { valueEncoding: 'json' });
+
+      // manually opening to fix https://github.com/Level/level/issues/221
+      await contractCache.open();
+
+      // Get keys that will be left, just to get the last one of them
+      const keys = await contractCache.keys({ reverse: true, limit: entriesStored }).all();
+      if (!keys || keys.length <= entriesStored) {
+        continue
+      }
+
+      contractCache.clear({ lt: keys[entriesStored - 1] })
+    }
+    return {
```

Comment:
please change the interface a bit so that the prune method would return `Promise<PruneState | null>`.

Null will simply mean that stats are not available.

Returning zeroes here might be misleading for the user (it may look as if the db was empty and nothing was removed;-))